include/swift/Runtime/RuntimeFunctions.def

```diff
@@ -638,7 +638,7 @@ FUNCTION(GetGenericMetadata, swift_getGenericMetadata,
 //                                     const OpaqueTypeDescriptor *descriptor,
 //                                     uintptr_t index);
 FUNCTION(GetOpaqueTypeMetadata, swift_getOpaqueTypeMetadata,
-         SwiftCC, OpaqueTypeAvailability,
+         SwiftCC, ConditionallyAvailable,
```

Comment:
These all still need to call a function to return the availability appropriate for the current deployment target. If we're deploying to an OS that always has Swift 5.1, these can all be `AlwaysAvailable`.

Reply:
This is what conditionally available means?

The code in getRuntimeFn() queries whether the feature is available based on whether something is AlwaysAvailable| ConditionallyAvailable | AvailableByCompatibilityLibrary:

```
  auto isFeatureAvailable = [&]() -> bool {
     auto deploymentAvailability =
         AvailabilityContext::forDeploymentTarget(*context);
     auto featureAvailability =();
     return deploymentAvailability.isContainedIn(featureAvailability);
   };

  switch (availability) {
   case RuntimeAvailability:: AlwaysAvailable:
     // Nothing to do.
     break;
   case RuntimeAvailability::ConditionallyAvailable: {
     isWeakLinked = !isFeatureAvailable();
     break;
   }
   case RuntimeAvailability::AvailableByCompatibilityLibrary: {
     if (!isFeatureAvailable())
       functionName.append("50");
     break;
   }
   }
```

In the future we can extend this by adding another entry to runtime.def that specifies at which swift version things became available ...

```
FUNCTION(GetOpaqueTypeMetadata, swift_getOpaqueTypeMetadata,
         SwiftCC, ConditionallyAvailable, Swift51
```


And then generate the availability based on that.
```
 context->get##AVAILVERSION##Availability
```

Reply:
An entry point is only `ConditionallyAvailable` if we're deploying back to a runtime version that may not have the entry point. If we're deploying to an OS with Swift 5.1, everything is `AlwaysAvailable`. Having a single version doesn't seem sufficient to me, because in the future, we may end up in a situation where a runtime entry point can only be back-deployed in the compatibility library to certain runtime versions; for instance, a Swift 5.2 feature may be completely unavailable on Swift 5.0, although we have a compatibility fallback that works for the 5.1 runtime. The most future-proof thing to do is to have the FUNCTION xmacro refer to a function that returns the enum value appropriate to the deployment target.

Reply:
I will address this in a follow-up patch.

Reply:
Attempt of follow-up here: https://github.com/apple/swift/pull/25573